### ISSUE TITLE
Incompatible types in assignment (expression has type "str", base class "SpectacularAPIView" defined the type as "None")

### ISSUE THREAD

**Issue by sshishov** [tfranzel/drf-spectacular#1120]
**Describe the bug**
When we are trying to set `api_verion` to some hardcoded version like `"v1"` we are getting error from mypy

**To Reproduce**
```python
from drf_spectacular import views as spectacular_views

class MyAPISchemaSpectacularAPIView(spectacular_views.SpectacularAPIView):
    api_version = 'v1'  # <-- this line is reported because `mypy` infer the type from baseclass as `NoneType`
```

**Expected behavior**
No error should be reported by mypy (v1.7.1).
By the way mypy (v1.6.2) did not have any issues with the code, maybe become more strict?

**Comment by tfranzel** [comment]
good point. turns out there are a bunch of other places, where this could become an issue. we have a PR open that fixes a heap of types but I never had the time to finish it. 

I will schedule some time to address the remaining type issues because the next release will be bigger anyways.

**Comment by sshishov** [comment]
Yep, sounds good: But even if you can release type imporovements slowly, also great 👍
